gateway/src/apicast/policy/soap/policy.lua

```diff
@@ -0,0 +1,68 @@
+--- SOAP Policy
+-- This policy adds support for a very small subset of SOAP.
+-- This policy basically expects a SOAPAction URI in the SOAPAction header or
+-- the content-type header.
+-- The SOAPAction header is used in v1.1:
+-- https://www.w3.org/TR/2000/NOTE-SOAP-20000508/#_Toc478383528, whereas the
+-- Content-Type header is used in v1.2:
+-- https://www.w3.org/TR/soap12-part2/#ActionFeature
+-- The SOAPAction URI is matched against the mapping rules defined for the
+-- service and the usage resulting from that is authorized and reported
+-- against 3scale's backend.
+
+local sub = string.sub
+local len = string.len
+
+local policy = require('apicast.policy')
+local _M = policy.new('SOAP policy')
+
+local soap_action_header = 'SOAPAction'
+local soap_action_ctype = 'application/soap+xml?action='
```

Comment:
I think this has to be parsed as URI args. Because we can't control the action appears right after as it can have `&` and be the last param.

Also quick google search shows some people use `;` to concat params. We should verify if that can really be the case.

Reply:
You're right @mikz . I wrongly assumed that `action` would be the only param.

Reply:
About using `;`, I think we need to support that: https://developer.mozilla.org/es/docs/Web/HTTP/Headers/Content-Type

Reply:
Yep. https://tools.ietf.org/html/rfc7231#section-3.1.1.1 says parameters are separated by `;` and  https://tools.ietf.org/html/rfc3902 says it is a parameter. I wonder where I've seen the example with `?`. Can't find it now.